shuffle_transfer failed during shuffle in `test_fused_blockwise_dataframe_merge`

**Describe the issue**:

Distributed test started failing in dask/dask CI with distributed `main`.

The error looks like this:

```
FAILED dask/tests/test_distributed.py::test_fused_blockwise_dataframe_merge[True] - RuntimeError: shuffle_barrier failed during shuffle 1d8d714896c0fa093a11a0ddfed35635
FAILED dask/tests/test_distributed.py::test_fused_blockwise_dataframe_merge[False] - RuntimeError: shuffle_transfer failed during shuffle 1d8d714896c0fa093a11a0ddfed35635
```

Example workflow:

https://github.com/dask/dask/actions/runs/5555724247/jobs/10147277695

**Anything else we need to know?**:

**Environment**:

- Dask version: 2023.7.0
- Python version:
- Operating System:
- Install method (conda, pip, source):

--- a/distributed/shuffle/_worker_plugin.py
+++ b/distributed/shuffle/_worker_plugin.py
@@ -951,7 +951,7 @@ def split_by_worker(
     # bytestream such that it cannot be deserialized anymore
     t = pa.Table.from_pandas(df, preserve_index=True)
     t = t.sort_by("_worker")
-    codes = np.asarray(t.select(["_worker"]))[0]
+    codes = np.asarray(t["_worker"])
     t = t.drop(["_worker"])
     del df
 
@@ -983,7 +983,7 @@ def split_by_partition(t: pa.Table, column: str) -> dict[Any, pa.Table]:
     partitions.sort()
     t = t.sort_by(column)
 
-    partition = np.asarray(t.select([column]))[0]
+    partition = np.asarray(t[column])
     splits = np.where(partition[1:] != partition[:-1])[0] + 1
     splits = np.concatenate([[0], splits])
 

